src/rogue/hardware/axi/AxiStreamDma.cpp

```diff
@@ -97,11 +98,18 @@ rha::AxiStreamDmaSharedPtr rha::AxiStreamDma::openShared(std::string path, rogue
       \t\taes-stream-driver = v5.16.0 (or later)\n\t\trogue = v5.13.0 (or later)"));
     }
 
+    // Check for mismatch in the rogue/loaded_driver API versions
+    if (dmaCheckVersion(ret->fd) < 0) {
+        ::close(ret->fd);
+        throw(rogue::GeneralError("AxiStreamDma::openShared",
+                                  "Rogue DmaDriver.h API Version (DMA_VERSION) does not match the aes-stream-driver API version"));
+    }
+
     // Result may be that rawBuff = NULL
     // Should this just be a warning?
     ret->rawBuff = dmaMapDma(ret->fd, &(ret->bCount), &(ret->bSize));
     if (ret->rawBuff == NULL) {
-        ret->bCount = ioctl(ret->fd, DMA_Get_Buff_Count, 0);
+        ret->bCount = dmaGetBuffCount(ret->fd);
```

Comment:
@slacrherbst Was there a "performance" reason why you were using a ioctl() in the AxiStreamDma.cpp code (instead of the header's API functions) here?  If yes, I will change it back